nemo/collections/nlp/data/machine_translation/preproc_mt_data.py

```diff
@@ -549,15 +584,28 @@ def train_yttm_tokenizers(
                         logging.info(
                             f'Encoder tokenizer model {encoder_tokenizer_model} not found. Training tokenizer model.'
                         )
-                        yttm.BPE.train(
-                            data=src_fname,
-                            vocab_size=encoder_tokenizer_vocab_size,
-                            model=encoder_tokenizer_model,
-                            coverage=encoder_tokenizer_coverage,
-                            n_threads=-1,
-                        )
+                        if encoder_tokenizer_name == "yttm":
+                            yttm.BPE.train(
+                                data=src_fname,
+                                vocab_size=encoder_tokenizer_vocab_size,
+                                model=encoder_tokenizer_model,
+                                coverage=encoder_tokenizer_coverage,
+                                n_threads=-1,
+                            )
+                        else:
+                            dir_name = os.path.dirname(encoder_tokenizer_model)
+                            create_spt_model(
+                                data_file=src_fname,
+                                vocab_size=encoder_tokenizer_vocab_size,
+                                sample_size=encoder_training_sample_size,
```

Comment:
Is the reason for specifying a sample_size here that sentencepiece cannot be trained on very large corpora?

Reply:
Yes. The variable though can be set to -1 (the default value) to fetch the entire data.